examples/music-player/src/components/AuthModal.tsx

```diff
@@ -0,0 +1,112 @@
+import { Button } from "@/components/ui/button";
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogHeader,
+  DialogTitle,
+} from "@/components/ui/dialog";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import { useAccount, usePasskeyAuth } from "jazz-react";
+import { Loader2 } from "lucide-react";
+import { useState } from "react";
+
+interface AuthModalProps {
+  open: boolean;
+  onOpenChange: (open: boolean) => void;
+}
+
+export function AuthModal({ open, onOpenChange }: AuthModalProps) {
+  const [username, setUsername] = useState("");
+  const [isSignUp, setIsSignUp] = useState(true);
+
+  const { me } = useAccount();
+
+  const [, authState] = usePasskeyAuth({
+    appName: "Jazz Music Player",
+    onAnonymousUserUpgrade: ({ username, isSignUp }) => {
+      if (isSignUp) {
+        me.profile!.name = username;
```

Comment:
Since profile is always available, shall we fix the type to reflect that? @aeplay